src/CHIPDriver.hh

```diff
@@ -118,4 +120,171 @@ extern std::atomic_ulong CHIPNumRegisteredFatBinaries;
  */
 extern thread_local hipError_t CHIPTlsLastError;
 
+class DeviceType {
+public:
+  enum Type { GPU, CPU, ACCEL, FPGA, DEFAULT };
```

Comment:
Nit: [Naming convention](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly).
```suggestion
  enum Type { GPU, CPU, Accelerator, FPGA, Default };
```